### DISCUSSION TITLE
Preventing nested functions

### DISCUSSION THREAD

**Discussion by Fakasht10** [MilesCranmer/PySR#322] [Q&A]
I Was wondering if it was possible to restrict the search to only f(t)=sin(t) and cos(t), functions and exclude function nesting, but can include powers of t such as t, t^2, sin (t^2)...basically
restrict ourselves to only include analytical functions and combination of functions from solutions to single degree of freedom vibration problems or vibration of rods and beams.
Would be very interested to hear your thoughts.
Respectfully
Aleksey

**Comment by MilesCranmer** [comment]
Hi Aleksey,
Thanks for the question. I think the `nested_constraints` parameter might be what you are looking for? https://astroautomata.com/PySR/api/#working-with-complexities Let me know if this answers it. Thanks!
Miles
